packets2/connack.go

```diff
@@ -0,0 +1,84 @@
+package packets2
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	pkts "github.com/energomonitor/bisquitt/packets"
+)
+
+const connackHeaderLength uint16 = 6
+const connackSessionPresentBit uint8 = 1
+
+type Connack struct {
+	pkts.Header
+	PacketV2
+	// Flags
+	SessionPresent bool
+	// Fields
+	SessionExpiryInterval    uint32
+	ReasonCode               ReasonCode
+	AssignedClientIdentifier string
+}
+
+func NewConnack(sessExpiry uint32, reasonCode ReasonCode, assignedClientID string,
+	sessPresent bool) *Connack {
+	p := &Connack{
+		Header:                   *pkts.NewHeader(pkts.CONNACK, 0),
+		SessionPresent:           sessPresent,
+		SessionExpiryInterval:    sessExpiry,
+		ReasonCode:               reasonCode,
+		AssignedClientIdentifier: assignedClientID,
+	}
+	p.computeLength()
+	return p
+}
+
+func (p *Connack) computeLength() {
+	assignedClientIDLength := uint16(len(p.AssignedClientIdentifier))
```

Comment:
I use `ClientIdentifier` only in the property name, to be consistent with the specification. Anywhere else I use `ClientID` because it's clear enough and shorter.

See e.g. https://github.com/energomonitor/bisquitt/blob/v2-connection/packets2/connack_test.go#L18

I think the inconsistency is acceptable here and I prefer it over literality here.

Reply:
Ok, fair enough.